azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -381,6 +428,21 @@ func resourceSubscriptionDelete(d *pluginsdk.ResourceData, meta interface{}) err
 		}
 	}
 
+	if d.HasChange("tags") {
+		tagsDetails := d.Get("tags").(map[string]interface{})
+		resource_tags := resources.Tags{
+			Tags: tags.Expand(tagsDetails),
+		}
+		locks.ByID(subscriptionId)
+		defer locks.UnlockByID(subscriptionId)
+		tagPatchParamter := resources.TagsPatchResource{Operation: "Delete", Properties: &resource_tags}
+		updateTags, delerr := tagsClient.UpdateAtScope(context.Background(), "subscriptions/"+subscriptionId, tagPatchParamter)
+		if delerr != nil {
+			return fmt.Errorf("Failed to Remove tags %q from subscription %q: %+v", tags.Flatten(resource_tags.Tags), subscriptionId, delerr)
+		}
+		d.Set("tags", *updateTags.ID)
+	}
+
```

Comment:
We shouldn't need to do anything with the tags here, as we will be disabling the subscription.
```suggestion
```